packages/scaffold-ui/src/views/w3m-switch-address-view/index.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  AccountController,
+  BlockchainApiController,
+  ModalController,
+  NetworkController,
+  OptionsController,
+  StorageUtil,
+  type AccountType
+} from '@web3modal/core'
+import { UiHelperUtil, customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import styles from './styles.js'
+
+@customElement('w3m-switch-address-view')
+export class W3mSwitchAddressView extends LitElement {
+  public static override styles = styles
+  // -- Members ------------------------------------------- //
+  private readonly metadata = OptionsController.state.metadata
+  public allAccounts: AccountType[] = AccountController.state.allAccounts || []
+  public readonly labels = AccountController.state.addressLabels
+  public readonly currentAddress: string = AccountController.state.address || ''
+  private balances: Record<string, number> = {}
+  private connectedConnector = StorageUtil.getConnectedConnector()
+  // Only show icon for AUTH accounts
+  private shouldShowIcon = this.connectedConnector === 'AUTH'
+  private caipNetwork = NetworkController.state.caipNetwork
+
+  constructor() {
+    super()
+    AccountController.subscribeKey('allAccounts', allAccounts => {
+      this.allAccounts = allAccounts
+      this.requestUpdate()
+    })
+  }
+
+  public override connectedCallback() {
+    super.connectedCallback()
+    this.allAccounts.forEach(account => {
+      BlockchainApiController.getBalance(account.address, this.caipNetwork?.id).then(response => {
+        let total = this.balances[account.address] || 0
+        if (response.balances.length > 0) {
+          total = response.balances.reduce((acc, balance) => acc + (balance?.value || 0), 0)
+        }
+        this.balances[account.address] = total
+        this.requestUpdate()
+      })
+    })
+  }
+
+  public getAddressIcon(type: AccountType['type']) {
+    if (type === 'smartAccount') {
+      return 'lightbulb'
+    }
+
+    return 'mail'
+  }
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    return html`
+      <wui-flex justifyContent="center" .padding=${['xl', '0', 'xl', '0'] as const}>
+        <wui-banner-img
+          imageSrc="${this.metadata?.icons[0]}"
+          text="${this.metadata?.url}"
+          size="sm"
+        ></wui-banner-img>
+      </wui-flex>
+      <wui-flex flexDirection="column" gap="xxl" .padding=${['l', 'xl', 'xl', 'xl'] as const}>
+        ${this.allAccounts.map(account => this.getAddressTemplate(account))}
+      </wui-flex>
+    `
+  }
+
+  // -- Private ------------------------------------------- //
+
+  private getAddressTemplate(account: AccountType) {
+    // If there is custom label for the address, use it
```

Comment:
Unless there is a complex logic / calculation, the code shouldn't require comments. The line is quite readable, I think we could remove the comment.